packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -0,0 +1,374 @@
+
+    //Tap center option
+    await tester.tap(find.byType(SegmentedControl));
+
+    //Shared value should equal center option's value
+    expect(sharedValue, widgets.elementAt(1).key);
+  });
+
+  testWidgets('Need at least 2 children', (WidgetTester tester) async {
+    final List<Widget> widgets = <Widget>[];
+    try {
+      await tester.pumpWidget(boilerplate(
+          child: SegmentedControl(
+              children: widgets, onValueChanged: (Key newValue) {})));
+      fail(
+          'Should not be possible to create a segmented control with no children');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children.length'));
+    }
+    try {
+      widgets
+          .add(const Text('Child 1', key: const ValueKey<String>('Child 1')));
+
+      await tester.pumpWidget(boilerplate(
+          child: SegmentedControl(
+              children: widgets, onValueChanged: (Key newValue) {})));
+      fail(
+          'Should not be possible to create a segmented control with just one child');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children.length'));
+    }
+  });
+
+  testWidgets('Children and onValueChanged can not be null',
+      (WidgetTester tester) async {
+    try {
+      await tester.pumpWidget(boilerplate(
+          child: SegmentedControl(
+              children: null, onValueChanged: (Key newValue) {})));
+      fail(
+          'Should not be possible to create segmented control with null children');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children'));
+    }
+
+    final List<Widget> widgets = <Widget>[];
+    widgets.add(const Icon(Icons.add, key: const ValueKey<String>('Add')));
+    widgets.add(const Icon(Icons.check, key: const ValueKey<String>('Check')));
+
+    try {
+      await tester.pumpWidget(boilerplate(
+          child: SegmentedControl(children: widgets, onValueChanged: null)));
+      fail(
+          'Should not be possible to create segmented control with null onValueChanged');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('onValueChanged'));
+    }
+  });
+
+  testWidgets(
+      'Widgets have correct default text/icon styles, change correctly on selection',
+      (WidgetTester tester) async {
+    final List<Widget> widgets = <Widget>[];
+    widgets.add(const Text('Child 1', key: const ValueKey<String>('Child 1')));
+    widgets.add(const Icon(Icons.add, key: const ValueKey<String>('Add')));
+    Key sharedValue = widgets.elementAt(0).key;
+
+    await tester.pumpWidget(new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+      return boilerplate(
+          child: SegmentedControl(
+              children: widgets,
+              onValueChanged: (Key newValue) {
+                setState(() {
+                  sharedValue = newValue;
+                });
+              },
+              value: sharedValue));
+    }));
```

Comment:
Try adding some commas between these multiple parenthesis and reformat to get a nice cascade.  Probably needed in a lot of places in this file.